### ISSUE TITLE
Uncaught TypeError: Cannot read property 'update' of undefined

### ISSUE THREAD

**Issue by silverspectro** [atom/atom#14950]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.19.0-beta2 ia32
**Electron**: 1.6.9
**OS**: Microsoft Windows 10 Famille
**Thrown From**: Atom Core

### Stack Trace

Uncaught TypeError: Cannot read property 'update' of undefined

```
At C:\Users\Ciro\AppData\Local\atom\app-1.19.0-beta2\resources\app\src\text-editor-component.js:3499

TypeError: Cannot read property 'update' of undefined
    at LinesTileComponent.updateLines (~/AppData/Local/atom/app-1.19.0-beta2/resources/app/src/text-editor-component.js:3499:28)
    at LinesTileComponent.update (~/AppData/Local/atom/app-1.19.0-beta2/resources/app/src/text-editor-component.js:3377:20)
    at updateComponent (~/AppData/Local/atom/app-1.19.0-beta2/resources/app/node_modules/etch/lib/patch.js:52:19)
    at patch (~/AppData/Local/atom/app-1.19.0-beta2/resources/app/node_modules/etch/lib/patch.js:16:25)
    at updateChildren (~/AppData/Local/atom/app-1.19.0-beta2/resources/app/node_modules/etch/lib/patch.js:77:13)
    at patch (~/AppData/Local/atom/app-1.19.0-beta2/resources/app/node_modules/etch/lib/patch.js:18:15)
    at updateChildren (~/AppData/Local/atom/app-1.19.0-beta2/resources/app/node_modules/etch/lib/patch.js:77:13)
    at patch (~/AppData/Local/atom/app-1.19.0-beta2/resources/app/node_modules/etch/lib/patch.js:18:15)
    at updateChildren (~/AppData/Local/atom/app-1.19.0-beta2/resources/app/node_modules/etch/lib/patch.js:77:13)
    at patch (~/AppData/Local/atom/app-1.19.0-beta2/resources/app/node_modules/etch/lib/patch.js:18:15)
    at updateChildren (~/AppData/Local/atom/app-1.19.0-beta2/resources/app/node_modules/etch/lib/patch.js:77:13)
    at patch (~/AppData/Local/atom/app-1.19.0-beta2/resources/app/node_modules/etch/lib/patch.js:18:15)
    at updateChildren (~/AppData/Local/atom/app-1.19.0-beta2/resources/app/node_modules/etch/lib/patch.js:77:13)
    at patch (~/AppData/Local/atom/app-1.19.0-beta2/resources/app/node_modules/etch/lib/patch.js:18:15)
    at Object.updateSync (~/AppData/Local/atom/app-1.19.0-beta2/resources/app/node_modules/etch/lib/component-helpers.js:108:26)
    at TextEditorComponent.updateSyncBeforeMeasuringContent (~/AppData/Local/atom/app-1.19.0-beta2/resources/app/src/text-editor-component.js:362:16)
    at TextEditorComponent.updateSync (~/AppData/Local/atom/app-1.19.0-beta2/resources/app/src/text-editor-component.js:259:16)
    at etch.getScheduler.updateDocument (~/AppData/Local/atom/app-1.19.0-beta2/resources/app/src/text-editor-component.js:212:46)
    at ViewRegistry.module.exports.ViewRegistry.performDocumentUpdate (~/AppData/Local/atom/app-1.19.0-beta2/resources/app/src/view-registry.js:187:15)
    at ~/AppData/Local/atom/app-1.19.0-beta2/resources/app/src/view-registry.js:3:65
```

### Commands

```
     -1:28.9.0 editor:newline (input.hidden-input)
     -1:26 core:backspace (input.hidden-input)
     -1:25.1.0 core:move-right (input.hidden-input)
     -1:24.5.0 core:save (input.hidden-input)
undefined -1:14.2.0 find-and-replace:select-next (input.hidden-input)
     -1:06.9.0 core:move-left (input.hidden-input)
  4x -1:06.8.0 core:move-right (input.hidden-input)
 96x -1:02.3.0 find-and-replace:select-next (input.hidden-input)
     -0:58.4.0 core:move-right (input.hidden-input)
     -0:58.2.0 core:move-left (input.hidden-input)
     -0:55 core:save (input.hidden-input)
     -0:35.9.0 core:copy (input.hidden-input)
  2x -0:10.5.0 editor:newline (input.hidden-input)
  2x -0:08.8.0 core:undo (input.hidden-input)
     -0:06.6.0 editor:newline (input.hidden-input)
     -0:06.2.0 core:undo (input.hidden-input)
```

### Non-Core Packages

```
atom-beautify 0.29.17 
atom-clock 0.1.5 
atom-minify 0.8.0 
atom-typescript 10.1.14 
autoclose-html 0.23.0 
autoprefixer 3.6.1 
color-picker 2.2.5 
language-elixir 0.17.0 
language-glsl 2.0.1 
linter 1.11.23 
linter-eslint 8.1.2 
minimap 4.26.8 
seti-syntax 1.1.2 
seti-ui 1.6.1 
```

**Comment by rsese** [comment]
Thanks for the report!  Can you confirm a few things for us?

* what were you doing at the time of the error?
* is the error is reproducible?
* if reproducible, do you see the error in safe mode (atom --safe)?

**Comment by Flascher** [comment]
I could not find reproducible conditions for this bug.
It happened to me when both undoing and redoing in a .scss file.

I tried multiple .scss files, it only occurred in one of them.

I tried other file extensions, no luck there either.

After saving my .scss file, the issue no longer occurs in that file either.

It may be worth noting that the hyperclick package was throwing errors simultaneously, specifically [this](https://github.com/facebooknuclide/hyperclick/issues/25) issue.

Sorry I couldn't be much more help here either.

**Atom**: 1.19.0-beta4 x64
**Electron**: 1.6.9
**OS**: Microsoft Windows 10 Pro N
**Thrown From**: Atom Core


### Stack Trace

Uncaught TypeError: Cannot read property 'update' of undefined

```
At C:\Users\alexa\AppData\Local\atom\app-1.19.0-beta4\resources\app\src\text-editor-component.js:3534

TypeError: Cannot read property 'update' of undefined
    at LinesTileComponent.updateLines (~/AppData/Local/atom/app-1.19.0-beta4/resources/app/src/text-editor-component.js:3534:28)
    at LinesTileComponent.update (~/AppData/Local/atom/app-1.19.0-beta4/resources/app/src/text-editor-component.js:3412:20)
    at updateComponent (~/AppData/Local/atom/app-1.19.0-beta4/resources/app/node_modules/etch/lib/patch.js:52:19)
    at patch (~/AppData/Local/atom/app-1.19.0-beta4/resources/app/node_modules/etch/lib/patch.js:16:25)
    at updateChildren (~/AppData/Local/atom/app-1.19.0-beta4/resources/app/node_modules/etch/lib/patch.js:77:13)
    at patch (~/AppData/Local/atom/app-1.19.0-beta4/resources/app/node_modules/etch/lib/patch.js:18:15)
    at updateChildren (~/AppData/Local/atom/app-1.19.0-beta4/resources/app/node_modules/etch/lib/patch.js:77:13)
    at patch (~/AppData/Local/atom/app-1.19.0-beta4/resources/app/node_modules/etch/lib/patch.js:18:15)
    at updateChildren (~/AppData/Local/atom/app-1.19.0-beta4/resources/app/node_modules/etch/lib/patch.js:77:13)
    at patch (~/AppData/Local/atom/app-1.19.0-beta4/resources/app/node_modules/etch/lib/patch.js:18:15)
    at updateChildren (~/AppData/Local/atom/app-1.19.0-beta4/resources/app/node_modules/etch/lib/patch.js:77:13)
    at patch (~/AppData/Local/atom/app-1.19.0-beta4/resources/app/node_modules/etch/lib/patch.js:18:15)
    at updateChildren (~/AppData/Local/atom/app-1.19.0-beta4/resources/app/node_modules/etch/lib/patch.js:77:13)
    at patch (~/AppData/Local/atom/app-1.19.0-beta4/resources/app/node_modules/etch/lib/patch.js:18:15)
    at Object.updateSync (~/AppData/Local/atom/app-1.19.0-beta4/resources/app/node_modules/etch/lib/component-helpers.js:108:26)
    at TextEditorComponent.updateSyncBeforeMeasuringContent (~/AppData/Local/atom/app-1.19.0-beta4/resources/app/src/text-editor-component.js:364:16)
    at TextEditorComponent.updateSync (~/AppData/Local/atom/app-1.19.0-beta4/resources/app/src/text-editor-component.js:260:16)
    at etch.getScheduler.updateDocument (~/AppData/Local/atom/app-1.19.0-beta4/resources/app/src/text-editor-component.js:213:46)
    at ViewRegistry.module.exports.ViewRegistry.performDocumentUpdate (~/AppData/Local/atom/app-1.19.0-beta4/resources/app/src/view-registry.js:187:15)
    at ~/AppData/Local/atom/app-1.19.0-beta4/resources/app/src/view-registry.js:3:65
```

### Commands

```
  4x -1:54.8.0 emmet:expand-abbreviation-with-tab (input.hidden-input)
     -1:50.9.0 core:backspace (input.hidden-input)
     -1:50.3.0 emmet:expand-abbreviation-with-tab (input.hidden-input)
  3x -1:49.8.0 core:move-up (input.hidden-input)
     -1:49.1.0 editor:move-to-first-character-of-line (input.hidden-input)
  4x -1:49 emmet:expand-abbreviation-with-tab (input.hidden-input)
     -1:44.1.0 core:save (input.hidden-input)
  2x -1:43.8.0 core:move-down (input.hidden-input)
  2x -1:40.9.0 core:select-up (input.hidden-input)
     -1:40 core:backspace (input.hidden-input)
  3x -1:34.2.0 core:select-up (input.hidden-input)
     -1:33.5.0 core:select-down (input.hidden-input)
  3x -1:33.1.0 core:backspace (input.hidden-input)
     -1:12.5.0 core:save (input.hidden-input)
  4x -1:10.3.0 core:undo (input.hidden-input)
     -1:06.4.0 core:save (input.hidden-input)
```

### Non-Core Packages

```
atom-beautify 0.30.4 
atom-material-syntax 1.0.6 
atom-material-ui 2.0.4 
atom-ternjs 0.18.3 
auto-detect-indentation 1.3.0 
autocomplete-modules 1.6.10 
browser-plus 0.0.98 
color-picker 2.2.5 
docblockr 0.11.0 
duotone-dark-amethyst-syntax 2.0.4 
duotone-dark-sea-syntax 2.1.0 
editorconfig 2.2.2 
emmet 2.4.3 
file-icons 2.1.9 
firewatch-syntax 1.1.1 
highlight-selected 0.13.1 
hyperclick 0.1.2 
js-hyperclick 1.11.0 
language-babel 2.67.2 
language-vue 0.23.0 
language-vue-component 0.5.0 
nucleus-dark-ui 0.12.3 
pigments 0.39.1 
todo-show 1.11.0 
vue-hyperclick 0.2.0 
vue2-autocomplete 0.1.0-alpha.4 
```

**Comment by as-cii** [comment]
It looks like this issue is related to https://github.com/facebooknuclide/hyperclick/issues/25 and, consequently, to https://github.com/facebooknuclide/hyperclick/issues/24, which has been fixed as part of https://github.com/atom/atom/pull/15051. 

The last report of this dates back to 1.19.0-beta4 so I wonder if it got fixed by the aforementioned pull-request (which landed on 1.19.0-beta5). Since we have no reproduction steps for this, what do you think about keeping this on our radar but revisit it once we get more reports?

**Comment by Ben3eeE** [comment]
I'm going to close this since it looks like it was fixed in beta and we have not had any more reports since :crossed_fingers:
